src/business-logic/services/inews-ftp-timestamp-parser-implementation.ts

```diff
@@ -0,0 +1,29 @@
+import { InewsFtpTimestampParser } from '../interfaces/inews-ftp-timestamp-parser'
+
+export class InewsFtpTimestampParserImplementation implements InewsFtpTimestampParser {
+  public parseInewsFtpTimestamp(timestampText: string): number {
+    const timePattern: RegExp = /(?<month>[a-z]+)\s+(?<dayOfMonth>[0-9]+)\s+((?<hours>[0-9]{2}):(?<minutes>[0-9]{2})|(?<year>[0-9]{4}))/i
+    const matchedGroup: Record<string, string> | undefined = timestampText.match(timePattern)?.groups
+    if (!matchedGroup || !matchedGroup['month'] || !matchedGroup['dayOfMonth']) {
+      return 0
+    }
```

Comment:
Based on feedback from Sonarlint, here is a suggestion for your consideration 😄 

```suggestion
const timePattern: RegExp = /(?<month>[a-z]+)\s+(?<dayOfMonth>\d+)\s+((?<hours>\d{2}):(?<minutes>\d{2})|(?<year>\d{4}))/i
    const matchedGroup: Record<string, string> | undefined = RegExp(timePattern).exec(timestampText)?.groups
    if (!matchedGroup?.month || !matchedGroup.dayOfMonth) {
      return 0
    }
```

Reply:
I have updated to use the suggestion.